github/actions/client_generate_jit_test.go

```diff
@@ -3,73 +3,60 @@ package actions_test
 import (
 	"context"
 	"net/http"
-	"net/http/httptest"
 	"testing"
 	"time"
 
 	"github.com/actions/actions-runner-controller/github/actions"
-	"github.com/google/go-cmp/cmp"
-	"github.com/hashicorp/go-retryablehttp"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
 func TestGenerateJitRunnerConfig(t *testing.T) {
-	token := "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwiaWF0IjoxNTE2MjM5MDIyLCJleHAiOjI1MTYyMzkwMjJ9.tlrHslTmDkoqnc4Kk9ISoKoUNDfHo-kjlH-ByISBqzE"
+	ctx := context.Background()
+	auth := &actions.ActionsAuth{
+		Token: "token",
+	}
 
 	t.Run("Get JIT Config for Runner", func(t *testing.T) {
-		name := "Get JIT Config for Runner"
 		want := &actions.RunnerScaleSetJitRunnerConfig{}
 		response := []byte(`{"count":1,"value":[{"id":1,"name":"scale-set-name"}]}`)
 
 		runnerSettings := &actions.RunnerScaleSetJitRunnerSetting{}
 
-		s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+		server := newActionsServer(t, http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
 			w.Write(response)
 		}))
-		defer s.Close()
+		client, err := actions.NewClient(ctx, server.configURLForOrg("my-org"), auth)
+		require.NoError(t, err)
 
-		actionsClient := actions.Client{
-			ActionsServiceURL:                 &s.URL,
-			ActionsServiceAdminToken:          &token,
-			ActionsServiceAdminTokenExpiresAt: &tokenExpireAt,
-		}
-
-		got, err := actionsClient.GenerateJitRunnerConfig(context.Background(), runnerSettings, 1)
-		if err != nil {
-			t.Fatalf("GenerateJitRunnerConfig got unexepected error, %v", err)
-		}
-
-		if diff := cmp.Diff(want, got); diff != "" {
-			t.Errorf("GenerateJitRunnerConfig(%v) mismatch (-want +got):\n%s", name, diff)
-		}
+		got, err := client.GenerateJitRunnerConfig(ctx, runnerSettings, 1)
+		require.NoError(t, err)
+		assert.Equal(t, want, got)
 	})
 
 	t.Run("Default retries on server error", func(t *testing.T) {
```

Comment:
Since the HTTP client is now persisted across requests, should we test that the client itself has retries or do we need to keep testing retries on all calls?

Reply:
I'd prefer the latter because we can ensure all calls use the retryable client or its equivalent!

Reply:
But given that the HTTP client is now re-used for the whole lifetime of an actions.Client wouldn't it make sense to have tests that verify that the option of having (or not) retries is observed? The only way to change the HTTP client down the line is for callers of this package to manually change `client.Client`, but then I would argue it's that caller to test responsibility that their override is working as intended. What do you think?

Reply:
Merging now to unlock other work on expanding the client, but definitely interested in this conversation!